### PR TITLE
Adding TS typings.

### PR DESCRIPTION
**Pull request by Fazendaaa**
Hey, @TooTallNate. Thanks for this awesome package!

Due to Hacktoberfest 2018, I'm contributing to some packages that I use; yours is one of those.

TypeScript typings help a lot of us who use it due to it's integration to TSLint and some more strict checkings, needed to be performed in some cases in the used packages as well; that's why I've decided to share the ones that I've made, making it more accessible to all that programs in this environment :)

Since I'm not altering any of your JavaScript code, didn't feel the need to write any test, and I only made typings for those methods flagged as public in your API JSDoc. But I ran the documentation examples in a dummy tests scenario through ts-node.

note: this is the project that I use your package, I intend to go back to it someday: https://github.com/Fazendaaa/ytlofi.

### CODE REVIEW

**Comment by TooTallNate** [comment]
Very cool! Thank you @Fazendaaa 👍

File: package.json [hunk 1]
```diff
@@ -16,6 +18,7 @@
     "readable-stream": "^2.3.3"
   },
   "devDependencies": {
+    "@machinomy/types-readable-stream": "git+https://github.com/machinomy/types-readable-stream.git",
```
Comment (LinusU) [inline package.json]: I don't see this module being actually used? 🤔

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+declare module 'speaker' {
```
Comment (LinusU) [inline index.d.ts]: You don't need this block. In fact, using it will declare a global module instead of describing the corresponding file (`index.js`)

**Comment by LinusU** [comment]
Awesome work 👏 

Left a few comments that would be nice to address in a follow up PR ☺️

File: package.json (hunk 1 above)
Comment (Fazendaaa) [reply package.json]: I thought that was the automatic import form **WritableOptions**... Now that you mention, I will remove it -- and fix a small typo in "types" that I only saw it now.

File: index.d.ts (hunk 2 above)
Comment (Fazendaaa) [reply index.d.ts]: Really? All of my TS typings I always did that way since I saw it how it's done. I remove it.

Comment (LinusU) [reply index.d.ts]: It's only needed when declaring types for another module, e.g. if you would publish a separate package with the types (`speaker-typings`) then you would have to have it ☺️

**Comment by LinusU** [comment]
🚢 Released in 0.4.2

**Pull request merged**